tests/state/ViewerState/constants.ts

```diff
@@ -1,7 +1,26 @@
-import { FeatureDataType } from "../../../src/colorizer";
+import { waitFor } from "@testing-library/react";
+import { act } from "react-dom/test-utils";
+
+import { Dataset, FeatureDataType } from "../../../src/colorizer";
 import { AnyManifestFile, ManifestFile } from "../../../src/colorizer/utils/dataset_utils";
+import { ViewerState } from "../../../src/state";
 import { DEFAULT_DATASET_DIR, makeMockDataset, MockArrayLoader, MockArraySource } from "../../test_utils";
 
+/**
+ * Wrapper around `store.setDataset()`. Allows for async operations to complete
+ * after a Dataset is set in the store.
+ */
+export const setDatasetAsync = async (
+  result: { current: ViewerState },
+  dataset: Dataset,
+  datasetKey: string = "some-dataset"
+): Promise<void> => {
+  act(() => {
+    result.current.setDataset(datasetKey, dataset);
+  });
+  await waitFor(() => {});
+};
```

Comment:
Needed to add this because the subscribers to Dataset now cross an async barrier. Any state updates that happen (even if they're instantaneous) will be treated by react-testing-library as happening outside of the `act` hook and it'll spit out warnings.